src/world/action/jump.c

```diff
@@ -1,15 +1,250 @@
+        temp_f2 = temp_v0->position.z;
+        D_8010C960 = temp_v0->position.x;
+        D_8010C97C = temp_f2;
+        func_802B6000_E24040();
+        playerStatus->flags |= 0x880000;
+        disable_player_input();
+    }
+
+    playerStatus->decorationList++;
+    if (playerStatus->fallState == 1) {
+        if ((playerStatus->flags & 0x80000000)) {
+
+          playerStatus->flags &= 0x7FFFFFF5;
+          playerStatus->flags |= 4;
+
+          phi_a0 = 0x6000A;
+
+          if (!(playerStatus->animFlags & PLAYER_ANIM_FLAG_3)) {
+            phi_a0 = 0x10008;
+          }
+
+          temp_s1 = phi_a0;
+          suggest_player_anim_clearUnkFlag(temp_s1);
+          cam = &gCameras;
+          cam->moveFlags |= 1;
+        }
+    } else {
+        return;
+    }
+    playerStatus->fallState++;
+}
+
+void func_802B6294_E242D4() {
+    s32 temp_v1;
+    s32 phi_a0;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Camera* cam;
+
+    if ((playerStatus->animFlags & PLAYER_ANIM_FLAG_PEACH_PHYSICS)) {
+        func_802B647C_E244BC();
+        return;
+    }
+
+    temp_v1 = playerStatus->flags;
+    if ( playerStatus->flags < 0) {
+        playerStatus->flags &= 0x7FFFFFF5;
+        playerStatus->flags |= 4;
+
+        if (playerStatus->animFlags & PLAYER_ANIM_FLAG_8BIT_MARIO) {
+            phi_a0 = 0x90005;
+        } else {
+            temp_v1 = 0x6000A;
+            phi_a0 = temp_v1;
+
+            if (!(playerStatus->animFlags & PLAYER_ANIM_FLAG_3)) {
+                phi_a0 = 0x10008;
+            }
+        }
+        suggest_player_anim_clearUnkFlag(phi_a0);
+        cam = &gCameras;
+        cam->moveFlags |= 1;
+    }
+    playerStatus->decorationList++;
+}
+
+void func_802B6348_E24388() {
+    f32* sub_arg1;
+    f32* sub_arg2;
+    f32* sub_arg3;
+    f32* sub_arg4;
+    f32 sub_arg5;
+    f32 sub_arg6;
+    f32 sub_arg7;
+    f32 sub_arg8;
+    f32 sp28;
+    f32 sp2C;
+    f32 sp30;
+    f32 sp34;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Camera* cam;
+
+    if (playerStatus->animFlags & PLAYER_ANIM_FLAG_PEACH_PHYSICS) {
+        func_802B6508_E24548();
+        return;
+    }
+
+    if (playerStatus->flags < 0) {
+        playerStatus->flags &= 0x7FFFFFF5;
+        playerStatus->flags |= 4;
+        cam = &gCameras;
+        cam->moveFlags |= 1;
+    }
+
+    playerStatus->decorationList++;
+    phys_update_interact_collider();
+    sub_arg1 = &sp28;
+    sub_arg2 = &sp2C;
+    sub_arg3 = &sp30;
+    sub_arg4 = &sp34;
+    sub_arg6 = sub_arg6;
+    sp28 = playerStatus->position.x;
+    sp2C = playerStatus->position.y;
+    sp30 = playerStatus->position.z;
+    sp34 = playerStatus->colliderHeight;
+
+    if (((get_collider_type_by_id(player_raycast_below_cam_relative(playerStatus, sub_arg1, sub_arg2, sub_arg3, sub_arg4,
+                                    &sub_arg5, &sub_arg6, &sub_arg7, &sub_arg8)) & 0xFF) - 2 >= 2u) && check_input_jump()) {
```

Comment:
Usually unsigned-literal comparisons like these are the result of gcc merging `||` operands into a more efficient instruction; not sure what `k - 2 >= 2U` would be but perhaps @ethteck has ideas

Reply:
I've never done one of these with a greater than ...huh. It's probably an or of some sort, but it'd require a temp and I'm not sure.......